core/relay/src/services/event_publish_service.rs

```diff
@@ -28,11 +34,334 @@ impl EventPublish for EventPublishService {
+        .map_err(|err| Error::GetQuery(format!("Failed to get event subscription state from db. Error: {:?}", err)))?;
+    
+    let event_publish_key = get_event_publication_key(request_id.to_string());
+    let event_id = Uuid::new_v4();
+    let target;
+    match state.clone().state {
+        Some(data) => match data {
+            view_payload::State::View(payload) => {
+                let request_state = RequestState {
+                    status: request_state::Status::EventReceived as i32,
+                    request_id: request_id.to_string(),
+                    state: Some(request_state::State::View(payload)),
+                };
+                target = EventState {
+                    state: Some(request_state),
+                    event_id: event_id.to_string(),
+                    message: "Successfully received state for event subscribed".to_string(),
+                };
+            }
+            view_payload::State::Error(error) => {
+                let request_state = RequestState {
+                    status: request_state::Status::Error as i32,
+                    request_id: request_id.to_string(),
+                    state: Some(request_state::State::Error(error)),
+                };
+                target = EventState {
+                    state: Some(request_state),
+                    event_id: event_id.to_string(),
+                    message: "Received error for the event subscribed".to_string(),
+                };
+            }
+        },
+        None => {
+            let request_state = RequestState {
+                status: request_state::Status::Error as i32,
+                request_id: request_id.to_string(),
+                state: Some(request_state::State::Error("Missing state".to_string())),
+            };
+            target = EventState {
+                state: Some(request_state),
+                event_id: event_id.to_string(),
+                message: "No state received for the event subscribed".to_string(),
+            };
+        }
+    };
+    let mut event_states_list: Vec<EventState> = Vec::new();
+    event_states_list.push(target.clone());
+    if db.has_key(event_publish_key.to_string())? {
+        let mut curr_event_states: EventStates = db.get::<EventStates>(event_publish_key.to_string()).expect("SendState: Unexpected DB Error");
+        event_states_list.append(&mut curr_event_states.states);
+    }
+    
+    let event_states: EventStates = EventStates {
+        states: event_states_list,
+    };
+    
+    db.set(&event_publish_key, &event_states)
+        .expect("Failed to insert into DB");
+    
+    spawn_handle_event(
+        state,
+        event_sub_state.event_publication_spec.expect("No Publication Spec stored in event subscription state"),
+        request_id,
+        event_id.to_string(),
+        conf
+    );
+    
+    return Ok(());
+}
+
```

Comment:
Can you please add comment that it will either call the APP URL or call the Driver based on the subscriber of the event.